### PR TITLE
[GEOT-6417]: Restoring deprecated predicate on ECQL

### PR DESCRIPTION
**Pull request by dromagnoli**

### CODE REVIEW

File: modules/library/cql/src/main/jjtree/ECQLGrammar.jjt [hunk 1]
```diff
@@ -138,15 +162,15 @@ TOKEN:
  */
 TOKEN [IGNORE_CASE]:  /* keywords */
 {
-   < AND: "and"> |
+   < AND: "and" | "&&"> |
    < OR: "or"> |
```
Comment (aaime) [inline modules/library/cql/src/main/jjtree/ECQLGrammar.jjt]: No "||" for or?

File: modules/library/cql/src/test/java/org/geotools/filter/text/ecql/ECQLComparisonPredicateTest.java [hunk 2]
```diff
@@ -57,6 +57,19 @@ public ECQLComparisonPredicateTest() {
         super(Language.ECQL);
     }
 
+    /**
+     * Equals predicate sample
+     *
+     * @throws Exception
+     */
+    @Test
+    public void deprecatedPredicate() throws Exception {
```
Comment (aaime) [inline modules/library/cql/src/test/java/org/geotools/filter/text/ecql/ECQLComparisonPredicateTest.java]: Should test all the added operators  and verify for reach that the result is the expected one

File: modules/library/cql/src/main/jjtree/ECQLGrammar.jjt (hunk 1 above)
Comment (dromagnoli) [reply modules/library/cql/src/main/jjtree/ECQLGrammar.jjt]: Nope. it wasn't part of the old CQLGrammar either.
Not sure why :)

File: modules/library/cql/src/test/java/org/geotools/filter/text/ecql/ECQLComparisonPredicateTest.java (hunk 2 above)
Comment (dromagnoli) [reply modules/library/cql/src/test/java/org/geotools/filter/text/ecql/ECQLComparisonPredicateTest.java]: Ok. I was just using the same tests available on the CQLComparisonPredicateTest.
I'm going to add new tests then.

Comment (dromagnoli) [reply modules/library/cql/src/test/java/org/geotools/filter/text/ecql/ECQLComparisonPredicateTest.java]: @aaime Tests added

**Comment by aaime** [comment]
Looks good thanks

**Pull request merged**